server-spi/src/main/java/org/keycloak/models/UserCredentialManager.java

```diff
@@ -26,9 +26,13 @@
 import java.util.stream.Stream;
 
 /**
+ * Manage the credentials for a user.
+ *
+ * @deprecated Instead of this class, use {@link UserModel#getUserCredentialManager()} instead.
  * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
  * @version $Revision: 1 $
  */
+@Deprecated
```

Comment:
Eventually yes, for now it will stay here as it is used by KeycloakSession via the `userCredentialManager()` method. Once that method is gone, this class can be moved.